### ISSUE TITLE
Denspose IUV map 

### ISSUE THREAD

**Issue by mithunlal** [facebookresearch/detectron2#1751]
## 🚀 Feature
A clear and concise description of the feature proposal.

## Motivation & Examples

Tell us why the feature is useful.

Describe what the feature would look like, if it is implemented.
Best demonstrated using **code examples** in addition to words.

## Note

We only consider adding new features if they are relevant to many users.

If you request implementation of research papers -- we only consider papers that have enough significance and prevalance in the object detection field.

We do not take requests for most projects in the `projects/` directory, because they are research code release that is mainly for other researchers to reproduce results.

"Make X faster/accurate" is not a valid feature request. "Implement a concrete feature that can make X faster/accurate" can be a valid feature request.

Instead of adding features inside detectron2,
you can implement many features by [extending detectron2](https://detectron2.readthedocs.io/tutorials/extend.html).
The [projects/](https://github.com/facebookresearch/detectron2/tree/master/projects/) directory contains many of such examples.

**Comment by jaymefosa** [comment]
seems this does the trick- let me know if this works for you:

```
sys.path.append(DenseposePath)

with open(dp_output, 'rb') as f:
    data = pickle.load(f)
    
i = data[0]['pred_densepose'][0].labels.cpu().numpy()
uv = data[0]['pred_densepose'][0].uv.cpu().numpy()
iuv = np.stack((uv[1,:,:], uv[0,:,:], i * 0,))
plt.imshow(np.transpose(iuv, (1,2,0)))

```
![image](https://user-images.githubusercontent.com/92468/124548654-f7c85b00-de68-11eb-952d-2c96e96165c8.png)

**Comment by jaymefosa** [comment]
slight adjustment:

```
i = data[0]['pred_densepose'][0].labels.cpu().numpy()
uv = data[0]['pred_densepose'][0].uv.cpu().numpy() * 255
iuv = np.stack((uv[1,:,:], uv[0,:,:], i))
iuv = np.transpose(iuv, (1,2,0))
plt.imshow(iuv / 255)

```
also make sure to save as .png

**Comment by Tangcj1226** [comment]
> 
> 
> slight adjustment:
> 
> ```
> i = data[0]['pred_densepose'][0].labels.cpu().numpy()
> uv = data[0]['pred_densepose'][0].uv.cpu().numpy() * 255
> iuv = np.stack((uv[1,:,:], uv[0,:,:], i))
> iuv = np.transpose(iuv, (1,2,0))
> plt.imshow(iuv / 255)
> ```
> 
> also make sure to save as .png

Is there a way to get the iuv_arr in the shape of the original image instead of the shape of the bounding box?I figured out how to get iuv map like you, but it's different from original shape. thanks.

**Comment by jaymefosa** [comment]
Yep, I don't have the code open in front of me but one of the keys in data is an [x, y, w, h] type array where the first two values are x1 and y1 for the upper left corner of detection, in the original image

**Comment by Tangcj1226** [comment]
> 
> 
> Yep, I don't have the code open in front of me but one of the keys in data is an [x, y, w, h] type array where the first two values are x1 and y1 for the upper left corner of detection, in the original image

thanks for your suggestion,i'll try

**Comment by Tangcj1226** [comment]
> 
> 
> seems this does the trick- let me know if this works for you:
> 
> ```
> sys.path.append(DenseposePath)
> 
> with open(dp_output, 'rb') as f:
>     data = pickle.load(f)
>     
> i = data[0]['pred_densepose'][0].labels.cpu().numpy()
> uv = data[0]['pred_densepose'][0].uv.cpu().numpy()
> iuv = np.stack((uv[1,:,:], uv[0,:,:], i * 0,))
> plt.imshow(np.transpose(iuv, (1,2,0)))
> ```
> 
> ![image](https://user-images.githubusercontent.com/92468/124548654-f7c85b00-de68-11eb-952d-2c96e96165c8.png)

supplement
you can get image in original shape by code:
box = data[0]["pred_boxes_XYXY"][0]
box[2]=box[2]-box[0]
box[3]=box[3]-box[1]
x,y,w,h=[int(v) for v in box]
bg=np.zeros((512,512,3))
bg[y:y+h,x:x+w,:]=iuv
plt.imshow(bg/255)

you can see more details in densepose.vis.base and densepose.vis.extracter
